tests/kernel/interrupt/src/dynamic_isr.c

```diff
@@ -50,10 +62,80 @@ void test_isr_dynamic(void)
 		     _sw_isr_table[i].arg == argval,
 		     "dynamic isr did not install successfully");
 }
+#else
+/*
+ * For testing arch such as x86, x86_64 and posix which support dynamic
+ * interrupt but without SW ISR table, we test it by applying for a dynamic
+ * interrupt and then trigger it to check if happened correctly.
+ */
+static int get_dynamic_interrupt_line(void)
+{
+#if defined(CONFIG_X86)
+	return TEST_IRQ_DYN_LINE;
+
+#elif defined(CONFIG_ARCH_POSIX)
+	hw_irq_ctrl_enable_irq(TEST_IRQ_DYN_LINE);
+	return TEST_IRQ_DYN_LINE;
+#else
+	return -1;
+#endif
```

Comment:
This can be simplified as
```
#ifdef TEST_IRQ_DYN_LINE
        return TEST_IRQ_DYN_LINE;
#else
        return -1;
#endif
```

Reply:
Thanks for this great comment, update it~